ios/brave-ios/Sources/Onboarding/WelcomeFocus/DefaultBrowserOnboardingStep.swift

```diff
@@ -25,7 +27,150 @@ struct DefaultBrowserGraphicView: View {
   }
 }
 
+@MainActor
+private final class DefaultBrowserPictureInPictureController: NSObject,
+  AVPictureInPictureControllerDelegate
+{
+  private let player: AVPlayer
+  private let playerLayer: AVPlayerLayer
+  private let controller: AVPictureInPictureController?
+  private let windowScene: UIWindowScene
+  private let containerView: UIView = .init()
+  private let onStop: () -> Void
+  private var didPlayToEndTime: NSObjectProtocol?
+  private var didBecomeActive: NSObjectProtocol?
+
+  init(
+    videoURL: URL,
+    windowScene: UIWindowScene,
+    onStop: @escaping () -> Void
+  ) {
+    #if DEBUG
+    assert(videoURL.isFileURL, "Video must be local")
+    assert(FileManager.default.fileExists(atPath: videoURL.path))
+    #endif
+    self.windowScene = windowScene
+    self.player = AVPlayer(playerItem: .init(asset: AVURLAsset(url: videoURL)))
+    self.playerLayer = AVPlayerLayer(player: self.player)
+    self.controller = AVPictureInPictureController(playerLayer: self.playerLayer)
+    self.onStop = onStop
+  }
+
+  func start() async {
+    guard let item = player.currentItem, let controller else { return }
+
+    player.actionAtItemEnd = .none
+    player.automaticallyWaitsToMinimizeStalling = false
+    playerLayer.videoGravity = .resizeAspect
+    playerLayer.frame = .init(width: 1, height: 1)  // Must have a non-zero size
+    containerView.alpha = 0
+    controller.requiresLinearPlayback = true
+    controller.delegate = self
+
+    // Repeat animation
+    didPlayToEndTime = NotificationCenter.default.addObserver(
+      forName: AVPlayerItem.didPlayToEndTimeNotification,
+      object: item,
+      queue: .main
+    ) { [player] _ in
+      MainActor.assumeIsolated {
+        player.seek(to: .zero)
+        player.play()
+      }
+    }
+
+    // Stop the PiP when the app comes back into focus
+    didBecomeActive = NotificationCenter.default.addObserver(
+      forName: UIApplication.didBecomeActiveNotification,
+      object: nil,
+      queue: .main
+    ) { [weak self] _ in
+      MainActor.assumeIsolated {
+        self?.stop()
+      }
+    }
+
+    // Add it to the view hierarchy which is required to start a PiP session
+    windowScene.keyWindow?.addSubview(containerView)
+    containerView.layer.addSublayer(playerLayer)
+
+    // Start playing
+    player.play()
+
+    // Wait until PiP is possible
+    if !controller.isPictureInPicturePossible {
+      await withCheckedContinuation { continuation in
+        var observer: NSKeyValueObservation?
+        observer = controller.observe(\.isPictureInPicturePossible, options: [.new]) { _, change in
+          _ = observer
+          observer = nil
+          if let isPossible = change.newValue, isPossible {
+            continuation.resume()
+          }
```

Comment:
Need to add a timeout here as its possible `isPictureInPicturePossible` never updates correctly on iPhone